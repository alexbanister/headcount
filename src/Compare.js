import React from 'react';
import CompareCard from './CompareCard';
import Card from './Card';
import PropTypes from 'prop-types';


const Compare = ( {compareInfo, removeCompare} ) => {
  let compareCard;
  let firstCard;
  let secondCard;
  if (compareInfo.firstDistrict) {
    firstCard = <Card
      setComparePosition={() => removeCompare(1)}
      selected='compare-card left'
      district={compareInfo.firstDistrict.districtData}
      key={`compare-${compareInfo.firstDistrict.title}`}/>;
  }
  if (compareInfo.firstDistrict && compareInfo.secondDistrict) {
    compareCard = <CompareCard
      compareInfo={compareInfo}
    />;
    secondCard = <Card
      setComparePosition={() => removeCompare(2)}
      selected='compare-card right'
      district={compareInfo.secondDistrict.districtData}
      key={`compare-${compareInfo.secondDistrict.title}`}/>;
  }
  return (
    <div className='compare-container'>
      {firstCard}
      {compareCard}
      {secondCard}
    </div>
  );
};

Compare.propTypes = {
  compareInfo: PropTypes.objectOf(PropTypes.object),
  removeCompare: PropTypes.func
};

export default Compare;
