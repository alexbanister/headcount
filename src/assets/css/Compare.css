.compare-container {
  background: #70879B;
  background: -moz-linear-gradient(top, #70879B 0%, #526B7F 100%);
  background: -webkit-linear-gradient(top, #70879B 0%, #526B7F 100%);
  background: linear-gradient(to bottom, #70879B 0%, #526B7F 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=0 );
  margin-bottom: 1em;
  display: flex;
  align-items: center;
  justify-content: center; }
  .compare-container .card-data, .compare-container h1 {
    display: flex;
    align-items: center;
    justify-content: center; }
  .compare-container h2:last-child {
    justify-content: flex-end; }

.compare-card {
  margin: 2em 0; }
  .compare-card .card-data {
    align-items: flex-start; }

.compare-card h2 {
  background: #F3A746;
  background: -moz-linear-gradient(top, #F3A746 0%, #EE7B2A 100%);
  background: -webkit-linear-gradient(top, #F3A746 0%, #EE7B2A 100%);
  background: linear-gradient(to bottom, #F3A746 0%, #EE7B2A 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=0 ); }
