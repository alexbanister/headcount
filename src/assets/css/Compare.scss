@import "variables.scss";

.compare-container {
  @include linear-gradient($select-color-dark, $select-color);
  margin-bottom: 1em;
  display: flex;
  align-items: center;
  justify-content: center;
  .card-data, h1 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
  h2:last-child {
    justify-content: flex-end;
  }
}

.compare-card {
  margin: 2em 0;
  .card-data {
    align-items: flex-start;
  }
}

.left h2 {
  @include linear-gradient($highlight-dark, $highlight);
}

.right h2 {
  @include linear-gradient($select-color-bright-shadow, $select-color-bright);
}
