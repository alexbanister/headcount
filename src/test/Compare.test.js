import React from 'react';
import { shallow } from 'enzyme';
import Compare from '../Compare';

describe('Compare', () => {

  it('should create an instance of a compare container', () => {
    const wrapper = shallow(<Compare
      compareInfo={{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
