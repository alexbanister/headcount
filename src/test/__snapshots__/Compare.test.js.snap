// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare should create an instance of a compare container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Compare
    compareInfo={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        undefined,
      ],
      "className": "compare-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          undefined,
        ],
        "className": "compare-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
